src/libs/actions/NameValuePair.js

```diff
@@ -37,7 +37,32 @@ function set(name, value, onyxKeyName) {
     }
 }
 
+/**
+ * Gets the value of an NVP
+ * migrationValueMap: {oldvalue: newvalue, ...}
+ *
+ * @param {String} name
+ * @param {String} onyxKey
+ * @param {*} [defaultValue]
+ * @param {Object} migrationValueMap
+ */
+function getAndMigrateValue(name, onyxKey, defaultValue, migrationValueMap) {
```

Comment:
What is this method for? If it's simply for migrating a value in Onyx, then a proper migration should be used rather than ALWAYS making this call from the router.

Reply:
I am migrating `FIRST_TIME_NEW_EXPENSIFY_USER_STEP: 'isFirstTimeNewExpensifyUser'` NVP
From boolean into number. I must set the value in server, so I add this function.
I am not sure, but what is better step in here?

Reply:
@tgolen the close library for migration with what I need is :
https://github.com/Expensify/App/blob/main/src/libs/migrations/RenamePriorityModeKey.js

Reply:
@tgolen the close library for migration with what I need is :

https://github.com/Expensify/App/blob/main/src/libs/migrations/RenamePriorityModeKey.js

It is for Onyx and it renames onyxkeys.

But for this problem, I need to change value of NVP. So it is stored in server.
I need the application to be authenticated first to retrieve NVP value, but onyx migration is called in here:

https://github.com/Expensify/App/blob/2c7c82cbe534e931b85a737a86aeb20bbef058ac/src/Expensify.js#L104

When application is launched and I think it is not yet authenticated. 

@tgolen Am I missing something here?

Reply:
Hm, OK I see what you are saying. I think maybe a better approach to this problem then is to do this:

1. Write code on the front-end that supports both NVP values (like a utility function)
2. Once that is deployed, we can run a query on the database to update the NVP value to be whatever new format is needed
3. Remove the code that supports both NVP values on the frontend.

Do you think that could work?

Reply:
@tgolen 

> 1. Write code on the front-end that supports both NVP values (like a utility function)

So I must consider the old value right? How about my function: 

```
function getAndMigrateValue(name, onyxKey, defaultValue, migrationValueMap) {
    API.Get({
        returnValueList: 'nameValuePairs',
        name,
    })
        .then((response) => {
            let value = lodashGet(response.nameValuePairs, [name], defaultValue);
            if (_.has(migrationValueMap, value)) {
                value = migrationValueMap[value];
                set(name, value, onyxKey);
            }
            Onyx.set(onyxKey, value);
        });
}
```

But I will remove ` set(name, value, onyxKey);` part right? It will be  handled in second step right?

> 2. Once that is deployed, we can run a query on the database to update the NVP value to be whatever new format is needed

This process in back end right?

> 3. Remove the code that supports both NVP values on the frontend.

Then the this `NameValuePair.getAndMigrateValue` function will be removed, and will use simple `NameValuePair.get`

<br />

Am I misunderstanding something here?